core/iwasm/common/wasm_memory.c

```diff
@@ -685,19 +699,30 @@ wasm_enlarge_memory_internal(WASMModuleInstance *module, uint32 inc_page_count)
     os_writegsbase(memory_data_new);
 #endif
 
+return_func:
+    if (!ret && enlarge_memory_error_cb)
+        enlarge_memory_error_cb(inc_page_count, total_size_old, 0,
+                                failure_reason, (wasm_module_inst_t)module,
+                                wasm_runtime_get_exec_env_singleton(
```

Comment:
Just found an issue: the exec_env returned by `wasm_runtime_get_exec_env_singleton` is not always the exec_env that current module instance uses (developer can also call `wasm_runtime_create_exec_env`), using it in the embedder may cause unexpected behavior.

There may be two possible ways: one is to add an extra argument `exec_env` in `wasm_enlarge_memory` and `aot_enlarge_memory`, and pass the correct exec_env from interpreter, fast-jit and AOT code.

The other is to add a field in module instance to store the current exec_env, e.g. WASMModuleInstanceExtra and AOTModuleInstanceExtra, set it every time in `wasm_call_function` and `aot_call_function`.

Maybe the latter is better as the former causes the ABI change of AOT code calling runtime API.

Reply:
> The other is to add a field in module instance to store the current exec_env, e.g. WASMModuleInstanceExtra and AOTModuleInstanceExtra, set it every time in wasm_call_function and aot_call_function.

Tried that, but something got broken for AOT (CI is failing).

Reply:
It seems to be due to `AOTModuleInstanceExtra` is different from `WASMModuleInstanceExtra`, could you see the comments and try again?

Reply:
Done, thanks